### ISSUE TITLE
Constant warning in Kubernetes

### ISSUE THREAD

**Issue by c-knowles** [containous/traefik#1462]
### What version of Traefik are you using (`traefik version`)?
1.2.1

### What is your environment & configuration (arguments, toml...)?
Kubernetes 1.6

### What did you do?
Deploy a service and ingress as per the documentation example.

### What did you expect to see?
Nothing above info level in the logs.

### What did you see instead?
Constant warning level logs of the form:
```
W  Service endpoints not found for REDACTED/REDACTED, falling back to Service ClusterIP 
W  Service endpoints not found for REDACTED/REDACTED, falling back to Service ClusterIP
W  Service endpoints not found for REDACTED/REDACTED, falling back to Service ClusterIP
W  Service endpoints not found for REDACTED/REDACTED, falling back to Service ClusterIP
W  Service endpoints not found for REDACTED/REDACTED, falling back to Service ClusterIP
```
One line roughly every second or so per service.

Seems to be caused by [this line](https://github.com/containous/traefik/blob/master/provider/kubernetes/kubernetes.go#L251). Given that `ClusterIP` is the default for a k8s Service plus it's what the documentation uses, does this need to be a warning? Perhaps lowering to debug or even removing would be appropriate.

**Comment by dtomcej** [comment]
Does the service have any actual endpoints? or are you just using an empty service?

**Comment by Regner** [comment]
For a bit of background as to why that is there: When you create a service in Kube it will have a list of endpoints. The endpoints are the pods that actually back the service. To help with performance Traefik attempts to skip a layer of proxying and just go straight to the pods themselves. So it listens to the Kube API and for each service it looks up the endpoints and then references them directly.

What this error really means is there are no running pods that match your services selector labels. Traefik falls back to using the service directly just incase for some reason that would work, but really it shouldn't.

**Comment by timoreimann** [comment]
@Regner If that's truly the case, then I'm very inclined to say we should drop this fallback-to-service logic. As you mentioned, it shouldn't work anyway; and if it somehow magically did, certain functions in Traefik would stop to work (like retrying to different Endpoints or sticky sessions). This would be a major confusion.

(Side note: Going straight for the Endpoints isn't a performance optimization: It's the way to implement typical proxy functions like sticky sessions or round-robin load-balancing, which you can't have if you work straight on the Service. Other HTTP ingress controllers work in the same fashion.)

I have another strong concern that the logic will mess with graceful termination where controllers rely on Endpoints being removed by kubelet once the ready state is marked as off. This signal should be respected by Traefik if we want to guarantee that no requests will be lost.

In summary, I'd rather skip Ingresses whose Endpoints are not yet available.

WDYT? @dtomcej? @errm?

**Comment by dtomcej** [comment]
@timoreimann is correct. If there are no endpoints listed, having the serviceIP does us no good. We should just disable this fallback. However, it does clearly log why traffic was not forwarded.

If we remove the fallback, we should still log that there were no available endpoints. We can't just skip it, because it is using state information (aka we are told that there should be a service with endpoints, just because there are no endpoints doesn't mean we can skip the service ;) ).

**Comment by c-knowles** [comment]
I had a look through which service were being logged by Traefik and in our case some of these were pods crashing, hence no endpoints. I have not determined the cause of the others as they are operational.

**Comment by timoreimann** [comment]
@dtomcej thanks, that does make sense.

Let's try to change the current behavior accordingly.

**Comment by timoreimann** [comment]
The erroneous fallback has been removed and error message improved in 1.3.0-rc3.
